great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -509,6 +513,26 @@ def get_domain_records(
         if TextClause and isinstance(selectable, TextClause):
             selectable = selectable.columns().subquery()
 
+        filter_conditions: List[RowCondition] = domain_kwargs.get(
+            "filter_conditions", []
+        )
+        # For SqlAlchemyExecutionEngine only one filter condition is allowed
+        if len(filter_conditions) == 1:
+            filter_condition = filter_conditions[0]
+            assert (
+                filter_condition.type_ == RowConditionParserType.GE
+            ), "filter_condition must be of type GE for SqlAlchemyExecutionEngine"
+
+            selectable = (
+                sa.select([sa.text("*")])
```

Comment:
@anthonyburdi a couple of things here: a) Do we want a type hint for selectable here?  And b) There might be a way to avoid `sa.text("*")` and use the more `SQLAlchemy`-like way -- if we want to pursue it, happy to discuss.

Reply:
a) good call, moved it up to the beginning of the method though since it is used in many places (this method could be split in a future PR).
b) Interesting - is it using the sa.Table object? Or something like `selectable = sa.select(selectable).where(parsed_condition)`?